### PR TITLE
Add debug logs

### PR DESCRIPTION
**Pull request by jehisonprada**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Summary

Add debug logs to track non paginated responses for endpoints that are supposed to return arrays with pagination

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [ ] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -149,17 +122,18 @@ export class ServiceNowClient {
...
-
         if (Array.isArray(result)) {
           const nextLink = getServiceNowNextLink(response?.headers?.link);
           return { result, nextLink };
+        } else {
+          // Temporary log to understand why we get non paginated responses for endpoint that are supposed to be paginated
+          this.logger.info(
+            {
+              resource: url,
+              schema: JSON.stringify(response?.data),
```
Comment (RonaldEAM) [inline src/client.ts:133]: We should not log the response data, it could have sensitive data from the customer

**Review by RonaldEAM — CHANGES_REQUESTED** [review]

File: src/client.ts (hunk 1 above)
Comment (RonaldEAM) [reply src/client.ts:133]: is there a specific key for the paginated fields?

Comment (jehisonprada) [reply src/client.ts]: Not really, the API responses always have a result key in place and that can be array or object depending on if the endpoint is or not paginated

File: src/client.ts [hunk 2]
```diff
@@ -217,24 +201,25 @@ export class ServiceNowClient {
...
     do {
-      const { result, nextLink } = await this.retryResourceRequest<
+      const paginatedResponse: PaginatedResponse<T> = await this.retryResourceRequest<
         PaginatedResponse<T>
       >(url);
 
-      if (Array.isArray(result)) {
-        for (const r of result) {
+      // For some reason Servicenow API sometimes have string responses for paginated endpoints
+      if (paginatedResponse.result && Array.isArray(paginatedResponse.result)) {
```
Comment (RonaldEAM) [inline src/client.ts:209]:
```suggestion
      if (paginatedResponse?.result && Array.isArray(paginatedResponse.result)) {
```

**Review by RonaldEAM — APPROVED** [review]

**Pull request merged**